src/java.base/share/classes/java/net/HttpURLConnection.java

```diff
@@ -606,7 +606,13 @@ public long getHeaderFieldDate(String name, long defaultValue) {
      * @return a {@code SocketPermission} object representing the
      *         permission necessary to connect to the destination
      *         host and port.
+     *
+     * @deprecated
+     * Permissions can no longer be used for controlling access to resources
+     * as the Security Manager is no longer supported.
      */
+    @Deprecated(since = "25", forRemoval = true)
+    @SuppressWarnings("removal")
```

Comment:
Do you still need this annotation now that `SocketPermission` is not deprecated for removal?

Reply:
Ah - good point - I will double check.

Reply:
Yes - it is needed as well. Apparently if you override a deprecated method you need both `@Deprecated` and `@SuppressWarning`:

```
src/java.base/share/classes/java/net/HttpURLConnection.java:615: warning: [removal] getPermission() in URLConnection has been deprecated and marked for removal
    public Permission getPermission() throws IOException {
                      ^
error: warnings found and -Werror specified
```

Reply:
Ok, but seems redundant.